### DISCUSSION TITLE
Input type=number data validation

### DISCUSSION THREAD

**Discussion by dsnair** [twilio-labs/paste#2502] [Q&A] [answered]
Hi, I noticed when I use the Input component with type number that I am still able to type non-numeric values. I see your [note](https://paste.twilio.design/components/input#number) that you don’t support HTML input[type="number"]. How else can I validate user entered data? Thanks for your answer!

**Comment by TheSisb** [comment, marked as answer]
Hi @dsnair,

Unfortunately, `<input type="number" />` doesn't prevent typing letters. In Firefox you can type any letter, and in Chrome you can still type `e`.  You can try it here: https://codesandbox.io/s/input-type-number-oqsybb?file=/src/App.js.  

From Paste's side of things, we setup `<Input type="number"` to correctly display the right settings for screen readers and mobile keyboards. Any input validation must be done by whoever implements our component, as it's hard to know how to validate generically. For example, currencies may be separated by `,`, `.`, ` `, or have no separation at the thousand mark. There are many little nuances to input validation. As such, it is impossible for us to do it better than product owners who know exactly what they need.

Hope this helps!
